src/common/SmartUrlInput.js

```diff
@@ -111,57 +115,48 @@ export default class SmartUrlInput extends PureComponent<Props, State> {
         }
       }, 100);
     }
-  };
+  }, []);
+
+  const renderPlaceholderPart = useCallback(
```

Comment:
Maybe skip the `useCallback` on this one. It's not any kind of event-handler callback -- rather it's a helper for rendering, to render a fragment of the element tree. And both the name, and the contents (consisting mostly of JSX, so producing some React elements to return), make it look very much like that's going to be the case.